spec/lib/publishing_api_payload_spec.rb

```diff
@@ -59,6 +59,24 @@
       expect(payload[:details][:change_history]).to match(history.change_history)
     end
 
+    it "delegates to PublishingApiPayload::FileAttachmentPayload to populate attachments" do
+      attachment = instance_double(
+        PublishingApiPayload::FileAttachmentPayload,
+        payload: {},
+      )
+
+      allow(PublishingApiPayload::FileAttachmentPayload).to receive(:new).and_return(attachment)
+
+      edition = build(
+        :edition,
+        file_attachment_revisions: [build(:file_attachment_revision)],
+      )
+
+      payload = described_class.new(edition).payload
+
+      expect(payload[:details][:attachments]).to match([attachment.payload])
+    end
```

Comment:
I think this test could actually be a bit shorter without using mocks - which could make it a little safer:

```
      file_attachment_revision = create(:file_attachment_revision)
      edition = build(:edition,
                      file_attachment_revisions: [file_attachment_revision])

      payload = described_class.new(edition).payload
      attachment_payload = PublishingApiPayload::FileAttachmentPayload
                             .new(file_attachment_revision, edition.document)
                             .payload

      expect(payload[:details][:attachments]).to contain_exactly(attachment_payload)
```